src/docfx/lib/JsonUtility.cs

```diff
@@ -342,10 +345,33 @@ private static bool CheckTypeContainsJsonExtensionData(Type type)
         {
             return s_cacheTypeContainsJsonExtensionData.GetOrAdd(
                 type,
-                new Lazy<bool>(() => type.GetProperties().Any(prop => prop.GetCustomAttribute<JsonExtensionDataAttribute>() != null))).Value;
+                new Lazy<bool>(() => type.GetProperties()
+                        .Any(prop => prop.GetCustomAttribute<JsonExtensionDataAttribute>() != null))).Value;
+        }
+
+        private static Type GetCollectionItemTypeFromArrayType(Type type)
+        {
+            var contract = DefaultDeserializer.ContractResolver.ResolveContract(type);
+            if (contract is JsonObjectContract)
+            {
+                return type;
+            }
+            else if (contract is JsonArrayContract arrayContract)
+            {
+                var itemType = arrayContract.CollectionItemType;
+                if (itemType is null)
+                {
+                    return type;
+                }
+                else
+                {
+                    return GetCollectionItemTypeFromArrayType(itemType);
```

Comment:
>return GetCollectionItemTypeFromArrayType(itemType); [](start = 20, length = 52)

As the outer `TraverseForUnknownFieldType` will be called recurtively, I think here only need to get the type of item of current array, which doens't need recursion to inspect deeper.